### PR TITLE
sigtool: 4a3719b4 -> 2a13539d

### PR DESCRIPTION
**Pull request by mstone**
###### Motivation for this change

This draft PR proposes updates to `darwin.sigtool` based on recent upstream development. It is being prepared in anticipation of a forthcoming upstream release (see thefloweringash/sigtool#2) and to facilitate broader testing of the new version against the rest of nixpkgs.

Key changes in this new version of `sigtool`: 

1. Replace the old sigtool `codesign.sh` wrapper script with a new `codesign` driver program implemented directly in C++, hopefully with good drop-in compatibility for various common uses of Apple's original `codesign` binary. 
2. Add limited support for Apple's "entitlements" system, which is used by, e.g., `qemu`, to access certain `*-darwin` features like `Hypervisor.framework`.
3. Various other contributed improvements.

CC: @thefloweringash 

Also potentially relevant to: #130132, #135877.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### Other notes

Since `sigtool` is used by `darwin.stdenv`, I anticipate that this or successor PRs will likely require a Darwin mass-rebuild.

### CODE REVIEW

File: pkgs/os-specific/darwin/sigtool/default.nix [hunk 1]
```diff
@@ -6,19 +6,12 @@ stdenv.mkDerivation {
   src = fetchFromGitHub {
     owner = "thefloweringash";
     repo = "sigtool";
-    rev = "4a3719b42dc91c3f513df94048851cc98e7c7fcf";
-    sha256 = "04ra1cx7k1sdbkj5yrvl0s3l333vpir8rnm8k1dh2zy1w0a6hpqa";
+    rev = "2a13539dc4893f39412a3fb810afc78b183df3df";
+    sha256 = "sha256-iCsdklN3crFx6CKsMIUP/fA3twLh4ArQh7OsVug5UjE=";
   };
 
   nativeBuildInputs = [ pkg-config makeWrapper ];
```
Comment (thefloweringash) [inline pkgs/os-specific/darwin/sigtool/default.nix:13]: `makeWrapper` is no longer required, don't forget to remove it from the imports too. Also on the topic of imports, `cmake` isn't used and should also be removed.

**Review by thefloweringash — COMMENTED** [review]
Thank you for picking this up. There's a little bit more tidy up we can do now that codesign is a regular binary and not a shell script. I don't know of a good way to share suggestions for changes outside of modified files, so here's my version as a gist to compare to: https://gist.github.com/thefloweringash/032517aec4e929c41a85efb24b33e16c

**Comment by mstone** [comment]
Hi @thefloweringash -- thanks enormously for the review above and for releasing sigtool-0.1.0; I've added three commits to address your review feedback and with that, unless other reviewers have additional suggestions, I think we're probably about ready to go here!

File: pkgs/os-specific/darwin/sigtool/default.nix [hunk 2]
```diff
@@ -1,24 +1,18 @@
...
+stdenv.mkDerivation rec {
+  pname = "sigtool";
+  version = "0.1.0";
 
   src = fetchFromGitHub {
     owner = "thefloweringash";
     repo = "sigtool";
-    rev = "4a3719b42dc91c3f513df94048851cc98e7c7fcf";
-    sha256 = "04ra1cx7k1sdbkj5yrvl0s3l333vpir8rnm8k1dh2zy1w0a6hpqa";
+    rev = "refs/tags/v${version}";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/sigtool/default.nix:10]:
```suggestion
    rev = "v${version}";
```

**Comment by alyssais** [comment]
@OfBorg build sigtool

**Comment by happysalada** [comment]
Unless I'm wrong the build is now failing on hydra
https://hydra.nixos.org/api/scmdiff?rev2=51bcdc4cdaac48535dabf0ad4642a66774c609ed&type=git&uri=https%3A%2F%2Fgithub.com%2FNixOS%2Fnixpkgs.git&rev1=a3a23d9599b0a82e333ad91db2cdc479313ce154&branch=

This is now a channel blocker for `unstable`.

The error is a little cryptic to me however. I'm happy to test and review if you have any clues as to what might be causing this.

**Comment by r-burns** [comment]
Fixup here: https://github.com/NixOS/nixpkgs/pull/139271

**Comment by thefloweringash** [comment]
> Fixup here: #139271

I had a different idea: #139272

**Comment by r-burns** [comment]
Aw, I didn't want to make you drop c++17 features. Filesystem works fine on clang 7, you just have to link the separate libc++fs. But we can do it whichever way you prefer.

**Comment by thefloweringash** [comment]
I tried bumping clang to clang_11 (with `stdenv = llvmPackages_11.stdenv`), and instead it fails with:
```
/tmp/nix-build-sigtool.drv-0/sigtool/commands.cpp:246:39: error: 'path' is unavailable: introduced in macOS 10.15
        identifier = std::filesystem::path(filename).filename();
```
Which made me think it would be easier to remove the one line that used it.

**Comment by r-burns** [comment]
I think you can work around this with _LIBCPP_DISABLE_AVAILABILITY but yeah maybe easier to just get rid of it.

**Pull request merged**